src/puppetlabs/puppetdb/config.clj

```diff
@@ -154,12 +155,24 @@
    :subprotocol "hsqldb"
    :subname     (format "file:%s;hsqldb.tx=mvcc;sql.syntax_pgs=true" (io/file vardir "db"))})
 
-(defn defaulted-db-connection
-  "Adds each of `hsql-default-connection` keypairs to `database` if classname, subprotocol and subname
-   are not found in the `database` config"
-  [{:keys [global database] :as config :or {database {}}}]
-  (cond-> config (not-any? database [:classname :subprotocol :subname])
-    (update :database merge (hsql-default-connection (:vardir global)))))
+(defn validate-db-settings
+  "Throws a {:type ::configuration-error :message m} exception
+  describing the required additions if the [database] configuration
+  doesn't specify classname, subprotocol and subname, all of which are
+  now required."
+  [{global :global db-config :database :or {db-config {}} :as config}]
+  (when (not-every? db-config [:classname :subprotocol :subname])
```

Comment:
should this be not-any?

As written if I omit one of these settings from my postgres config, I get the hsqldb configuration error. On master if I do that it fails with "missing required key: classname".  If I omit all of them, it starts me on hsqldb. Seems like that behavior is preferable.